lib/Sema/LookupVisibleDecls.cpp

```diff
@@ -463,12 +475,17 @@ static void lookupDeclsFromProtocolsBeingConformedTo(
           };
           DeclVisibilityKind ReasonForThisDecl = ReasonForThisProtocol;
           if (const auto Witness = NormalConformance->getWitness(VD)) {
-            if (Witness.getDecl()->getName() == VD->getName()) {
+            auto *WD = Witness.getDecl();
+            if (WD->getName() == VD->getName()) {
               if (LS.isIncludingDerivedRequirements() &&
                   Reason == DeclVisibilityKind::MemberOfCurrentNominal &&
-                  isDerivedRequirement(Witness.getDecl())) {
+                  isDerivedRequirement(WD)) {
                 ReasonForThisDecl =
                     DeclVisibilityKind::MemberOfProtocolDerivedByCurrentNominal;
+              } else if (!LS.isIncludingProtocolExtensionMembers() &&
+                         WD->getDeclContext()->getExtendedProtocolDecl()) {
+                // Don't skip this requiement.
+                // Witnesses in protocol extensions aren't reported.
```

Comment:
Could you given an example of what this is doing? I'm having trouble following.

Reply:
`lookupVisibleMemberDecls()` generally prefer witness members over the requirements. For example
```swift
protocol P {
  func foo()
}
extension P {
 func foo() {}
}
struct S: P {
  #^COMPLETE^#
}
```
It skips the requirement and report `P.foo()` *from the extension* which will be collected in `doGlobalExntensionLookup()` call below.
In "ignoring protocol extension" mode, since we don't collect decls from protocol extensions, we need to report the requirement itself here.

Reply:
Ah, that makes sense, thanks! Could you add this comment, maybe above the line that does "continue"?  E.g.

```
// lookupVisibleMemberDecls() generally prefers witness members over requirements.
continue;
```

I think that would be enough to make it clear for future readers.

Reply:
Sure. Done.